blog/README.md

```diff
@@ -0,0 +1,232 @@
+depmod -F /lib/modules/4.18.0-372.53.1.el8_6.x86_64/System.map 4.18.0-372.53.1.el8_6.x86_64
+--> 26dfedc51be
+[2/2] STEP 1/5: FROM quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:127670aaa6073e97bd6d919e1b57bd16b60435410b096aeefce7cd64f6c83d24
+[2/2] STEP 2/5: ARG KERNEL_VERSION
+--> 2879805a870
+[2/2] STEP 3/5: COPY --from=builder /etc/driver-toolkit-release.json /etc/
+--> 8dea5bb2bd8
+[2/2] STEP 4/5: COPY --from=builder /lib/modules/4.18.0-372.53.1.el8_6.x86_64/*.ko /usr/lib/modules/4.18.0-372.53.1.el8_6.x86_64/
+--> 6249995c281
+[2/2] STEP 5/5: RUN depmod -a 4.18.0-372.53.1.el8_6.x86_64 && echo simple_kmod > /etc/modules-load.d/simple_kmod.conf
+[2/2] COMMIT quay.io/ybettan/coreos-layering:simple-kmod
+--> 612c8c8a54f
+Successfully tagged quay.io/ybettan/coreos-layering:simple-kmod
+612c8c8a54f0800cf8dcb4c20d5d3d384dc47463711d6d3ad1bd93d52e982b1e
+
+```
+
+Now, let's push the image to some image registry in order for the image to be accesible
+by the cluster.
+```
+$ podman push quay.io/ybettan/coreos-layering:simple-kmod
+```
+
+Make sure the image is public or it have the correct pull-secret to pull the image.
+The pull secret in the cluster is
+```
+$ oc get secret/pull-secret -n openshift-config.
+```
+
+### Loading the kernel modules into the nodes
+
+First, we will need to create a `MachineConfig` file
+```
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 99-ybettan-external-image
+spec:
+  osImageURL: quay.io/ybettan/coreos-layering:simple-kmod
+```
+
+Make sure to change the `osImageURL` section in the MachineConfig yaml to the
+image you just built, then apply it.
+```
+$ oc apply -f machineconfig.yaml
+```
+
+The [machine-config-operator](https://github.com/openshift/machine-config-operator)
+will start updating workers nodes one by one with the new image specified in
+`osImageURL` which contain our kernel module.
+
+This may take some time and you can track the progress using
+```
+$ watch oc get nodes
+```
+until all workers nodes are `Ready` again.
+
+Here is a snapshot of the node updating process
+```
+NAME                           STATUS                     ROLES                  AGE   VERSION
+ip-10-0-146-108.ec2.internal   Ready                      control-plane,master   39m   v1.25.8+37a9a08
+ip-10-0-147-43.ec2.internal    Ready,SchedulingDisabled   worker                 32m   v1.25.8+37a9a08
+ip-10-0-166-128.ec2.internal   Ready                      control-plane,master   39m   v1.25.8+37a9a08
+ip-10-0-178-9.ec2.internal     Ready                      worker                 30m   v1.25.8+37a9a08
+ip-10-0-226-15.ec2.internal    Ready                      control-plane,master   39m   v1.25.8+37a9a08
+ip-10-0-228-190.ec2.internal   Ready                      worker                 29m   v1.25.8+37a9a08
+```
+
+### Validate that the kernel module was loaded
+
+This can be achieved simple by
```

Comment:
lets add a command to show that the node is using a layering rather than the original installation

Reply:
You mean showing the `.ko` file on the host? or somehow using `rpm-ostree` to validate this is a layered image?

Reply:
I have added the output of `rpm-ostree status` running on the node to show that the node is running a custome image - https://github.com/ybettan/coreos-layering-driver-toolkit/pull/4/files#diff-78f236b87da98227ae670c36caae7662a3f9e664737f5a8feecdbd5d8b71545aR225-R237.